src/mlpack/tests/main_tests/preprocess_split_test.cpp

```diff
@@ -0,0 +1,190 @@
+                      ("test_labels").n_cols,
+                      std::floor(0.1 * labels_size));
+}
+
+/**
+ * Ensure that test ratio is always a non-negative number.
+ */
+BOOST_AUTO_TEST_CASE(PreprocessSplitTestRatioTest)
+{
+  // Load custom dataset.
+  arma::mat inputData;
+  arma::Mat<size_t> labels;
+  data::Load("vc2.csv", inputData);
+  data::Load("vc2_labels.txt", labels);
+
+  // Input custom data points and labels.
+  SetInputParam("input", std::move(inputData));
+  SetInputParam("input_labels", std::move(labels));
+
+  SetInputParam("test_ratio", (double) -0.2);
+
+  Log::Fatal.ignoreInput = true;
+  BOOST_REQUIRE_THROW(mlpackMain(), std::runtime_error);
+  Log::Fatal.ignoreInput = false;
+}
+
+/**
+ * Check that if test size is 0 then
+ * train consist of whole input data.
+ */
+BOOST_AUTO_TEST_CASE(PreprocessSplitZeroTestRatioTest)
+{
+  // Load custom dataset.
+  arma::mat inputData;
+  arma::Mat<size_t> labels;
+  data::Load("vc2.csv", inputData);
+  data::Load("vc2_labels.txt", labels);
+
+  // Store size of input dataset.
+  int input_size  = inputData.n_cols;
+  int labels_size  = labels.n_cols;
+
+  // Input custom data points and labels.
+  SetInputParam("input", std::move(inputData));
+  SetInputParam("input_labels", std::move(labels));
+
+  SetInputParam("test_ratio", (double) 0.0);
+
+  mlpackMain();
+
+  // Now check that the output has desired dimensions.
+  BOOST_REQUIRE_EQUAL(CLI::GetParam<arma::mat>("training").n_cols,
+                      input_size);
+  BOOST_REQUIRE_EQUAL(CLI::GetParam<arma::mat>("test").n_cols,
+                      0);
+
+  BOOST_REQUIRE_EQUAL(CLI::GetParam<arma::Mat<size_t>>
+                      ("training_labels").n_cols, labels_size);
+  BOOST_REQUIRE_EQUAL(CLI::GetParam<arma::Mat<size_t>>
+                      ("test_labels").n_cols, 0);
+}
+
+/**
+ * Check that if test size is 1 then
+ * test consist of whole input data.
+ */
+BOOST_AUTO_TEST_CASE(PreprocessSplitUnityTestRatioTest)
+{
+  // Load custom dataset.
+  arma::mat inputData;
+  arma::Mat<size_t> labels;
+  data::Load("vc2.csv", inputData);
+  data::Load("vc2_labels.txt", labels);
+
+  // Store size of input dataset.
+  int input_size  = inputData.n_cols;
+  int labels_size  = labels.n_cols;
+
+  // Input custom data points and labels.
+  SetInputParam("input", std::move(inputData));
+  SetInputParam("input_labels", std::move(labels));
+
+  SetInputParam("test_ratio", (double) 1.0);
+
+  mlpackMain();
+
+  // Now check that the output has desired dimensions.
+  BOOST_REQUIRE_EQUAL(CLI::GetParam<arma::mat>("training").n_cols,
+                      0);
+  BOOST_REQUIRE_EQUAL(CLI::GetParam<arma::mat>("test").n_cols,
+                      input_size);
+
+  BOOST_REQUIRE_EQUAL(CLI::GetParam<arma::Mat<size_t>>
+                      ("training_labels").n_cols, 0);
+  BOOST_REQUIRE_EQUAL(CLI::GetParam<arma::Mat<size_t>>
+                      ("test_labels").n_cols, labels_size);
+}
+
```

Comment:
These tests look good; my only suggestion for another test is one where labels are not given.  It's possible to use `mlpack_preprocess_split` while only specifying a matrix with data, and this should work fine also.